spec/controllers/bills_controller_spec.rb

```diff
@@ -0,0 +1,72 @@
+require 'rails_helper'
+
+describe BillsController do
+
+  before(:each) { @bill = create(:bill) }
+
+  describe 'GET #index' do
+    before { get :index }
+
+    it { is_expected.to respond_with :ok }
+    it { is_expected.to render_template :index }
+    it { expect(@bill).to_not be nil }
+  end
+
+  describe "GET #new" do
+    before { get :new }
+
+    it { is_expected.to respond_with :ok }
+    it { is_expected.to render_template :new }
+  end
+
+  describe "POST #create" do
+
+    it "should create bill" do
+      user = create(:user)
+      sign_in user
+      bill = FactoryGirl.attributes_for(:bill).merge(user_id: user.id, units: {number_occupants: 4})
+
+      expect{ post(:create, bill: bill) }.to change{ Bill.count }.by(1)
+    end
+  end
+
+  describe "GET #edit" do
+    before { get :edit, id: @bill }
+
+    it { is_expected.to respond_with :ok }
+    it { is_expected.to render_template :edit }
+  end
+
+
+  describe "GET #comarison" do
```

Comment:
1. `comarison` -> `comparison`
2. Generally, we should limit our controller actions to the standard 7 for CRUD. Otherwise, this controller could get unwieldy really quickly. I suggest that we have a `BillComparison` controller (as a [singular resource], perhaps), move this over there, and make it a `show` action that doesn't have to take an id.

[singular resource]: http://guides.rubyonrails.org/routing.html#singular-resources